### ISSUE TITLE
MetricsQL: Missing results with certain regex patterns

### ISSUE THREAD

**Issue by jduncan0000** [VictoriaMetrics/VictoriaMetrics#2255]
**Describe the bug**
When running a query with an empty `|` filter at the end of a regex expression, results are inconsistently returned, depending on the other filters specified. 

Ex. 

Note: In this example, `type` is not a label on `vm_app_version`, but it is a label on other unrelated metrics. 

`{__name__=~"vm_app_v.*",type=""}`
returns expected results (metrics meeting the name filter criteria that do not have a `type` label)

`{__name__=~"vm_app_v.*",type=~"foo|"}`
returns no results (not expected - expected same as above)

`{__name__=~"vm_app_.*",type=~"foo|"}`
returns expected results (metrics meeting the name filter criteria that do not have a `type` label)

`{__name__=~"vm_app_v.*",bar=~"foo|"}`
returns expected results (metrics meeting the name filter criteria that do not have a `bar` label)

**To Reproduce**
I have been trying to nail down the exact set of criteria that results in this behavior but I am not 100% sure. It seems to happen in some (but not all) cases with the following conditions:

1. Regex matching on a label that does not exist for the metric using the empty pipe convention `foo=~"bar|"`
2. `foo` is a label that exists on other metrics but not the one(s) being queried for

In the example above, it also only happens when ` __name__` is filtered by regex down to a certain point, but I have also observed this behavior when specifying a metric name without a __name__ matcher ( ex. `metric_name{status="foo|"}` ) in other cases.

**Expected behavior**
`{__name__=~"vm_app_v.*",type=~"foo|"}` should return the same results as `{__name__=~"vm_app_v.*",type=""}`

**Logs**
None

**Screenshots**
`{__name__=~"vm_app_v.*",type=""}`
returns expected results (metrics meeting the name filter criteria that do not have a `type` label)
![image](https://user-images.githubusercontent.com/3580723/155895094-dcaf6978-28fd-4ea5-a008-e699331ef540.png)

`{__name__=~"vm_app_v.*",type=~"foo|"}`
returns no results (not expected - expected same as above)
![image](https://user-images.githubusercontent.com/3580723/155895118-947ed568-c543-4046-8d78-93a906529e4f.png)

`{__name__=~"vm_app_.*",type=~"foo|"}`
returns expected results (metrics meeting the name filter criteria that do not have a `type` label)
![image](https://user-images.githubusercontent.com/3580723/155895133-1b84a09a-5181-4492-a719-e537a99f2611.png)

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
$ ./victoria-metrics-prod --version
victoria-metrics-20220222-193240-tags-v1.73.1-0-gacbea6c1e
```

**Used command-line flags**
Default

**Comment by f41gh7** [comment]
I've tried to reproduce it, but for me it works. Looks like some flaky bug.

**Comment by jduncan0000** [comment]
@f41gh7 Thanks for taking a look - I did some debugging on an instance where I can reproduce the issue and submitted a PR based on my findings. This change fixed the issue for me, but let me know what you think. Thanks again!

**Comment by valyala** [comment]
The related issue - https://github.com/VictoriaMetrics/VictoriaMetrics/issues/546

**Comment by valyala** [comment]
The [pull request](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/2304), which fixes the issue, is included in [v1.75.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.

@jduncan0000 , thanks for discovering and fixing the issue!
